### DISCUSSION TITLE
Is a `dhdl.xvg` file needed with `.collect()` and `.analyze()`?

### DISCUSSION THREAD

**Discussion by a-ws-m** [Becksteinlab/MDPOW#274] [Q&A]
The [docs suggest ](https://mdpow.readthedocs.io/en/latest/init.html#analyze-output-and-logpow-calculation) using `.collect()` and `.analyze()`, but there's a couple of issues as far as I can tell:

1. The default `Gsolv.setup()`, using `edr=True`, sets the `.mdp` option `separate-dhdl-file = no`. There are therefore no `.xvg` files produced.
2. The `dgdl_xvg()` function looks for a file whose name is `<deffnm>.xvg`, swapping out the bit in tags, but the default name for the file produced when using `-dhdl` is `dhdl.xvg`.

I've switched to using the `alchemlyb` implementation of these functions, but it would be good to know if I was doing something wrong and to update the above example if not.

PS I can see that `analyze()` handles the `IOError` thrown by `dgdl_xvg()` and looks for the `.edr` file instead. Maybe I'm not supposed to call `.collect()` before `.analyze()`?

**Comment by orbeckst** [comment]
The fact that we default to EDR and then `collect()` only looks for XVG seems to indicate that the docs have not been kept up-to date or that the code has simply become inconsistent. It would make more sense if `collect()` looked for EDR files first. The whole logic in `analysis()` in the block https://github.com/Becksteinlab/MDPOW/blob/120c5dddd67d14c493971125e6355414b09e73c1/mdpow/fep.py#L1089C1-L1097C26
```python
            try:
                self.collect(stride=stride, autosave=False)
            except IOError as err:
                if err.errno == errno.ENOENT:
                    self.convert_edr()
                    self.collect(stride=stride, autosave=False)
                else:
                    logger.exception(err)
                    raise
```
would be better in `collect()`. Basically, 
1. check which format files are present, XVG or EDR?
2. if EDR, convert them to xvg.bz2
3. then parse everything 

In short, the easiest thing would be to remove the `collect()` call from the docs because `analyze()` does it better at the moment.

Ultimately, exactly the same thing happens with `collect_alchemlyb()` (can only use XVG and fails if there's none) and `analyze_alchemlyb()`.


(Sidenote: A while back we switched to using EDR files by default because they are more compact. Unfortunately, we still do what seems a waste, namely, converting EDR free energy data to XVG because we haven't implemented an actually EDR reader (with [`pyedr`](https://github.com/MDAnalysis/panedr)).

It would also be _really_ good if someone found the time to add a EDR reader to alchemlyb https://github.com/alchemistry/alchemlyb/issues/67 because then we wouldn't need to do the conversion.)
